primary/src/tests/block_remover_tests.rs

```diff
@@ -459,21 +474,24 @@ async fn test_unlocking_pending_requests() {
 
 #[must_use]
 pub fn worker_listener(
+    keypair: NetworkKeyPair,
     address: multiaddr::Multiaddr,
     expected_batch_ids: Vec<BatchDigest>,
     tx_delete_batches: metered_channel::Sender<DeleteBatchResult>,
 ) -> JoinHandle<()> {
-    println!("[{}] Setting up server", &address);
-    let mut recv = PrimaryToWorkerMockServer::spawn(address.clone());
+    let pubkey = keypair.public().clone();
+    println!("[{}] Setting up server", pubkey);
+    let (mut recv, network) = PrimaryToWorkerMockServer::spawn(keypair, address);
     tokio::spawn(async move {
+        let _network = network;
```

Comment:
What's this for? is it some kind of closure capture thing? worth explaining or is it idiomatic and I just don't know Rust well enough yet :)

Reply:
Yeah this is just so that the network handle is kept alive, since it will shutdown the network once all handles have been dropped.